corehq/apps/userreports/sql/adapter.py

```diff
@@ -124,8 +124,11 @@ def delete(self, doc):
             connection.execute(delete)
 
     def doc_exists(self, doc):
-        query = self.get_query_object().filter_by(doc_id=doc['_id'])
-        return self.session_helper.Session.query(query.exists()).scalar()
+        try:
+            query = self.get_query_object().filter_by(doc_id=doc['_id'])
+            return self.session_helper.Session.query(query.exists()).scalar()
+        finally:
+            self.session_helper.Session.commit()
```

Comment:
I think using this context manager would better: https://github.com/dimagi/commcare-hq/blob/25a3e34c7ef995879562c5b77f4767fa80ceb4a4/corehq/apps/callcenter/queries.py#L20-L20